apps/studio/components/grid/components/formatter/ForeignKeyFormatter.tsx

```diff
@@ -14,19 +14,18 @@ import { NullValue } from '../common/NullValue'
 import { ReferenceRecordPeek } from './ReferenceRecordPeek'
 
 interface Props extends PropsWithChildren<RenderCellProps<SupaRow, unknown>> {
-  tableId?: string
+  tableId?: number
 }
 
 export const ForeignKeyFormatter = (props: Props) => {
   const { project } = useProjectContext()
 
   const { tableId, row, column } = props
-  const id = tableId ? Number(tableId) : undefined
 
   const { data } = useTableEditorQuery({
     projectRef: project?.ref,
     connectionString: project?.connectionString,
-    id,
+    id: tableId,
```

Comment:
sorry if i might be overlooking something - but wondering if we could just use the table id from the URL via useParams actually? or is the tableId here potentially not the table that the user is looking at?

Reply:
I think this would probably be alright to switch, but I'm gonna err on the side of caution and leave it as a prop – since it's just the one prop!